### PR TITLE
Using gmt in NixOS

### PR DESCRIPTION
**Pull request by miniBill**
Adding NixOS instructions

### CODE REVIEW

File: doc/rst/source/install.rst [hunk 1]
```diff
@@ -278,6 +278,15 @@ Install GMT via:
 
    sudo emerge --verbose --ask sci-geosciences/gmt
 
+NixOS
+~~~~~
+
+Use GMT in your shell via:
```
Comment (joa-quim) [inline doc/rst/source/install.rst:284]: I guess you mean "Install GMT ..."?

Comment (miniBill) [reply doc/rst/source/install.rst:284]: No, I intentionally wrote it that way.

`nix shell nixpkgs#packagename` will start a shell where `packagename` is available without the need to permanently install it on your system (when you close the shell, the package is not used/active anymore and is going to be eventually garbage collected).

To install it you would either add it to your system configuration or your home-manager configuration. Instructions on how to do that are probably unneeded for most NixOS users, the important part is the list of packages.

Comment (miniBill) [reply doc/rst/source/install.rst:284]: You can also install the packages via `nix-env -iA`, but that's nondeclarative and honestly better not used at all in favor of a fully declarative approach.

Comment (joa-quim) [reply doc/rst/source/install.rst:284]: OK, that sounds wasteful but I know nothing about NixOS (other than the name). But this is a section about how to install GMT. Wouldn't it be appropriate to add that info on NixOS installation as well?

Comment (miniBill) [reply doc/rst/source/install.rst:284]: The installation instructions would be "add `gmt`, `gshhg-gmt` and `dcw-gmt` to your list of packages". I can add that, I guess.

Comment (joa-quim) [reply doc/rst/source/install.rst:284]: Yes, please. And explain also that difference between using and installing. NixOS users may know that but others likely do not.

Comment (miniBill) [reply doc/rst/source/install.rst]: Done

Comment (joa-quim) [reply doc/rst/source/install.rst]: I mean, You can leave both informations. `Using` and `Installing` if you find it appropriate.

Comment (miniBill) [reply doc/rst/source/install.rst]: Nah, this is fine

**Review by joa-quim — APPROVED** [review]

**Pull request merged**